app/src/main/java/org/xbmc/kore/host/HostConnectionObserver.java

```diff
@@ -477,6 +548,27 @@ public void onError(int errorCode, String description) {
         }, checkerHandler);
     }
 
+    private void checkPlaylist() {
+        connection.execute(new GetPlaylist(connection), new ApiCallback<ArrayList<GetPlaylist.GetPlaylistResult>>() {
+            @Override
+            public void onSuccess(ArrayList<GetPlaylist.GetPlaylistResult> result) {
+                if (result.isEmpty())
+                    return;
+
+                for (PlaylistEventsObserver observer : playlistEventsObservers) {
+                    observer.playlistsAvailable(result);
+                }
+            }
+
+            @Override
+            public void onError(int errorCode, String description) {
+                for (PlaylistEventsObserver observer : playlistEventsObservers) {
+                    observer.playlistOnError(errorCode, description);
+                }
+            }
+        }, new Handler());
+    }
+
```

Comment:
Edit: i'm not saying that we should definitely store and check the previous results, just that we should evaluate if that's feasible, and if not, what are the consequences (observers being continuously notified, even without any changes, which causes them to update the UI unnecessarily).

Reply:
I'll have some more comments on the `HostConnection.execute(Callable)`, but here i'd just like to point out that, unlike in other notifications, we are not storing the previous result, and only passing the notification if something changes.
So, we are always notifying observers of `playlistsAvailable`/`playlistOnError` - every 2 seconds if through HTTP. A notification should only be sent when something changes - because it makes sense conceptually, and it minimizes observers work, for instance if they are always getting the `somethingIsPlaying` notification and redrawing the UI it can cause flicker - , and that's why in the other cases we're saving and checking the previous results. I'm not sure if that can be done with Playlists, but we should be aware of this.